Src/WitsmlExplorer.Api/Services/CredentialsCache.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.Caching;
+
+using Microsoft.Extensions.Logging;
+namespace WitsmlExplorer.Api.Services
+{
+
+    public interface ICredentialsCache
+    {
+        void Set(string cacheId, string encryptedCredentials, CacheItemPolicy policy);
+        public string Get(string cacheId);
+        public long Count();
+        public void RefreshSession(string cacheId, string encryptedCredentials);
+        public void RemoveAllClientCredentials(string clientId);
+    }
+
+    public class CredentialsCache : ICredentialsCache
+    {
+        private readonly ObjectCache _cache = MemoryCache.Default;
+        private readonly ILogger<CredentialsCache> _logger;
+
+        public CredentialsCache(ILogger<CredentialsCache> logger)
+        {
+            _logger = logger;
+        }
+
+        public void Set(string cacheId, string encryptedCredentials, CacheItemPolicy policy)
+        {
+            _cache.Set(cacheId, encryptedCredentials, policy);
+        }
+
+        public string Get(string cacheId)
+        {
+            return _cache.Get(cacheId) as string;
+        }
+        public long Count()
+        {
+            return _cache.GetCount();
+        }
+
+        public void RefreshSession(string cacheId, string encryptedCredentials)
```

Comment:
The RefreshSession method seems to be unused, should it be removed? Also it seems to replicate the [SlidingExpiration policy](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.caching.cacheitempolicy.slidingexpiration?view=dotnet-plat-ext-7.0#system-runtime-caching-cacheitempolicy-slidingexpiration) .